node/src/actors/chain_manager/mod.rs

```diff
@@ -1295,6 +1300,11 @@ impl ChainManager {
                         &act.chain_state.alt_keys,
                     );
 
+                    // Put the local superblock into chain state
+                    act.chain_state
+                        .superblock_state
+                        .set_current_superblock(superblock.clone());
```

Comment:
Can we move this logic inside `act.chain_state.superblock_state.build_superblock`?

Reply:
What's the advantage? My concern is that `build_superblock` looks like it's modeled in a rather self-contained way and it felt strange to me if it made any significant changes on `self.chain_state`.

Reply:
The advantage if we merge them is that we can be sure that `superblock_state.get_current_superblock()` returns the last superblock. Because the only way to change it is using `superblock_state.build_superblock()`. By adding a `set_current_superblock` method we must make sure that it is always called after `superblock_state.build_superblock()`, which is error prone. It's a similar argument to making fields public vs private.